File: front/ebs-frontend/src/services/ReaderService.js
import axios from "axios";
import { REST_API_BASE_URL } from '../../config'

const SERVICE_PATH = REST_API_BASE_URL + 'readers/'

export const listReaders = () => axios.get(SERVICE_PATH + 'all');

export const createReader = (reader) => axios.post(SERVICE_PATH + 'create', reader)

export const getReader = (readerId) => axios.get(SERVICE_PATH + 'get/' + readerId)

export const updateReader = (readerId, reader) => axios.put(SERVICE_PATH + 'update/' + readerId, reader)

export const deleteReader = (readerId) => axios.delete(SERVICE_PATH + 'delete/' + readerId)

File: ebs/src/main/kotlin/com/bft/ebs/services/impl/ReaderService.kt
package com.bft.ebs.services.impl

import com.bft.ebs.dto.ReaderDto
import com.bft.ebs.repositories.impl.ReaderRepository
import com.bft.ebs.services.EbsMapper
import com.bft.ebs.services.interfaces.ReaderService
import org.springframework.stereotype.Service

@Service
class ReaderService(private val readerRepository: ReaderRepository) :
    ReaderService {
    override fun getReader(id: Long): ReaderDto {
        return EbsMapper.mapReaderToReaderDto(readerRepository.read(id))
    }

    override fun getAllReaders(): List<ReaderDto> {
        return readerRepository.findAll().map { EbsMapper.mapReaderToReaderDto(it) }
    }

    override fun createReader(readerDto: ReaderDto): Long {
        return readerRepository.create(
            EbsMapper.mapReaderDtoToReader(readerDto)
        ).id ?: throw IllegalStateException("Couldn't create the reader'")
    }

    override fun deleteReader(id: Long): Boolean {
        return readerRepository.delete(id)
    }

    override fun updateReader(readerDto: ReaderDto): Boolean {
        return readerRepository.update(EbsMapper.mapReaderDtoToReader(readerDto))
    }
}